### DISCUSSION TITLE
Fetch is sometimes not starting/making progress

### DISCUSSION THREAD

**Discussion by cvoica** [nats-io/nats.go#1167] [Q&A]
I'd like to ask you for ideas on how to investigate a situation I'm encountering with Fetch(). 

I create several connections to the NATS server and on each connection I do the following:
- create a durable consumers
- PullSubscribe()
- Fetch 1000 messages with 60s timeout + send the messages to another processing goroutine via a channel
- process those messages + ack 

When I start a few connection (e.g. 10) most of the consumers are making progress but a few of them are stuck, the cli command: nats con info shows, for example: Last Delivered Message: Consumer sequence: 0 Stream sequence: 384,706,024

I checked the goroutine stack and it looks to me as Fetch is not getting any messages, it's just stuck
Any ideas how to investigate this?

I use this version
    github.com/nats-io/jsm.go v0.0.35
    github.com/nats-io/nats.go v1.21.0

**Comment by derekcollison** [comment]
Best to share a gist of your actual code.

/cc @piotrpio @wallyqs
